pkg/instance/instance.go

```diff
@@ -163,13 +163,14 @@ func (i *Instance) VFS() vfs.VFS {
 
 func (i *Instance) makeVFS() error {
 	fsURL := config.FsURL()
+	mutex := vfs.NewMemLock()
```

Comment:
As I understand, it creates a new mutex for each request. So, two requests for the same domain will use two different locks, can mess things. I think we should have a map for domain -> mutex (its-self protected by a mutex)